lib/core/src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -370,13 +369,13 @@ prefilterBlock b u0 = runState $ do
         state (isOurs acct) <&> \case
             Nothing -> Nothing
             Just{}  -> Just (acct, amt)
-    mkTxMeta :: Natural -> Direction -> TxMeta
-    mkTxMeta amt dir = TxMeta
+    mkTxMeta :: Coin -> Direction -> TxMeta
+    mkTxMeta amount dir = TxMeta
         { status = InLedger
         , direction = dir
         , slotNo = b ^. #header . #slotNo
         , blockHeight = b ^. #header . #blockHeight
-        , amount = Quantity amt
+        , amount = amount
```

Comment:
This should be okay-ish for Cardano because we know that the max supply is 42B. But we should generally be careful when summing token quantities. Even if each token individually fit in a Word64, their sum might not. So for anything that regards folds or sums, we should stick to `Natural` IMO.

Reply:
At the very least, if changing this to `Coin`, I'd like to leave a note in the code about exactly this.

Reply:
I have added a note to `Coin.hs`.

In `Model.hs`, any sums are limited to a single transaction, so it's all fine for ada amounts, and token amounts use Natural.

I would not be against changing newtype Coin to use Natural or Integer, just to remove all doubt about overflow.